confluent_kafka/schema_registry/avro.py

```diff
@@ -0,0 +1,227 @@
+import json
+import struct
+from copy import deepcopy
+
+from fastavro import parse_schema, schemaless_writer, schemaless_reader
+
+from confluent_kafka.schema_registry import MessageField, TopicNameStrategy, _MAGIC_BYTE, \
+    SubjectNameStrategy, ContextStringIO
+from confluent_kafka.serialization import Serializer, Deserializer, SerializationError
+
+
+class AvroSerializer(Serializer):
+    """
+    AvroSerializer encodes objects into the Avro binary format in accordance
+    with the schema provided at instantiation.
+
+    Unless configured otherwise the AvroSerializer will automatically register
+    its schema with the provided registry instance.
+
+    Args:
+        conf (dict): Avro Serializer configuration instance.
+        schema_registry_client (SchemaRegistryClient): Schema Registry client.
+        schema (Schema): Schema instance to use when encoding Avro records.
```

Comment:
you should at least check what's passed in has type avro. currently thinking it's better to just pass in schema_str because the other fields are useless (and people may think they are relevant). might be seeing that the wrong way though.